### PR TITLE
Replace BookRepository with JpaRepository add liquebase support, add …

### PR DESCRIPTION
**Pull request by OlegTuryk**
…softdelete concept

### CODE REVIEW

File: src/main/java/com/app/onlinebookstore/specifications/BookSpecifications.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import com.app.onlinebookstore.dto.BookSearchParametersDto;
+import com.app.onlinebookstore.model.Book;
+import jakarta.persistence.criteria.Predicate;
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.data.jpa.domain.Specification;
+
+public class BookSpecifications {
+    public static Specification<Book> searchBooks(BookSearchParametersDto searchParameters) {
+        return (root, query, cb) -> {
```
Comment (DmytroV95) [inline src/main/java/com/app/onlinebookstore/specifications/BookSpecifications.java]: I think it's a more understandable name of argument.
```suggestion
        return (root, query, criteriaBuilder) -> {
```

File: src/main/java/com/app/onlinebookstore/specifications/BookSpecifications.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            List<Predicate> predicates = new ArrayList<>();
+            if (searchParameters.getTitle() != null) {
+                predicates.add(cb.equal(root.get("title"), searchParameters.getTitle()));
+            }
+            if (searchParameters.getAuthor() != null) {
+                predicates.add(cb.equal(root.get("author"), searchParameters.getAuthor()));
```
Comment (DmytroV95) [inline src/main/java/com/app/onlinebookstore/specifications/BookSpecifications.java]: It's better to make "author"  and other values as constants.

**Review by DmytroV95 — CHANGES_REQUESTED** [review]
GJ! Interesting solution. Let's improve it!

**Review by DmytroV95 — APPROVED** [review]
GJ!

**Review by vovasalyha — APPROVED** [review]
GJ.

**Review by andrii-hoienko — COMMENTED** [review]
GJ!

**Pull request merged**